From d6db871542b7c8bc019fb7a45d958ce82cc9fc48 Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Thu, 11 Sep 2014 12:00:07 -0400
Subject: [infiniband] mlx5: Expose support for signature MR feature

Message-id: <b02f8f56efa1a08b36f061b72928cf3b017a4782.1410425016.git.avadai@redhat.com>
Patchwork-id: 93655
O-Subject: [Patch RHEL7.1 V1 071/197] IB/mlx5: Expose support for signature MR feature
Bugzilla: 1107617 1107618 1107619 1107620
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Sagi Grimberg <sagig@mellanox.com>

Currently support only T10-DIF types of signature handover operations
(types 1|2|3).

Signed-off-by: Sagi Grimberg <sagig@mellanox.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit 2dea909444c294f55316c068906945ef38980ef3)
Signed-off-by: Amir Vadai <avadai@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/mlx5/main.c b/drivers/infiniband/hw/mlx5/main.c
index 3258d4d..fa6dc87 100644
--- a/drivers/infiniband/hw/mlx5/main.c
+++ b/drivers/infiniband/hw/mlx5/main.c
@@ -273,6 +273,15 @@ static int mlx5_ib_query_device(struct ib_device *ibdev,
  if (flags & MLX5_DEV_CAP_FLAG_XRC)
   props->device_cap_flags |= IB_DEVICE_XRC;
  props->device_cap_flags |= IB_DEVICE_MEM_MGT_EXTENSIONS;
+ if (flags & MLX5_DEV_CAP_FLAG_SIG_HAND_OVER) {
+  props->device_cap_flags |= IB_DEVICE_SIGNATURE_HANDOVER;
+  /* At this stage no support for signature handover */
+  props->sig_prot_cap = IB_PROT_T10DIF_TYPE_1 |
+          IB_PROT_T10DIF_TYPE_2 |
+          IB_PROT_T10DIF_TYPE_3;
+  props->sig_guard_cap = IB_GUARD_T10DIF_CRC |
+           IB_GUARD_T10DIF_CSUM;
+ }
 
  props->vendor_id    = be32_to_cpup((__be32 *)(out_mad->data + 36)) &
   0xffffff;
-- 
1.7.1